br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -481,6 +489,12 @@ func (m *DupeDetector) RecordDataConflictError(ctx context.Context, stream DupKV
 		if err != nil {
 			return errors.Trace(err)
 		}
+
+		if algorithm == config.DupeResAlgErr {
+			err = common.ErrFoundDuplicateKeys.FastGenByArgs(key, val)
+			return ConvertToErrFoundConflictRecords(err, tbl)
```

Comment:
here we can directly use key and val. No need to create a temporary err variable. You can add another function like NewErrFoundConflictRecords(key, value, tbl) and let ConvertToErrFoundConflictRecords use it internally.